internal/engine/wazevo/ssa/pass.go

```diff
@@ -309,26 +311,15 @@ func (b *builder) clearBlkVisited() {
 
 // passNopInstElimination eliminates the instructions which is essentially a no-op.
 func passNopInstElimination(b *builder) {
-	if int(b.nextValueID) >= len(b.valueIDToInstruction) {
-		b.valueIDToInstruction = append(b.valueIDToInstruction, make([]*Instruction, b.nextValueID)...)
-	}
+	ensureValueIdToInstructionInit(b)
```

Comment:
on second thought, it looks like it has to be reset each time (and/or we must ensure it is consistent across passes)

Reply:
not it shouldn't be necessary to do each time on each pass - should be fine with valueIDToInstruction once.

Reply:
I wonder if it makes sense to refactor the remove the instructions in `passDeadCodeEliminationOpt` and invoke this instead.

Reply:
> on second thought, it looks like it has to be reset each time (and/or we must ensure it is consistent across passes)

> no it shouldn't be necessary to do each time on each pass

I got misled because passes are also invoked individually. So, I refactored this init procedure in its own pass ( `passCollectValueIdToInstructionMapping`) which is expected to execute before the dependant passes (nopInstElim and constFold for now).